File: jarvis_brain.js
const axios = require('axios');
require('dotenv').config();
const { loadMemory } = require('./conversation_memory');

async function askJarvis(prompt) {
    const memory = loadMemory();
    const recent = memory.slice(-5);

    const messages = [
        { role: "system", content: "You are JARVIS, Ra's helpful assistant." },
        ...recent.flatMap(m => [
            { role: "user", content: m.user },
            { role: "assistant", content: m.jarvis }
        ]),
        { role: "user", content: prompt }
    ];

    const response = await axios.post(
        'https://api.openai.com/v1/chat/completions',
        {
            model: "gpt-3.5-turbo",
            messages
        },
        {
            headers: {
                Authorization: `Bearer ${process.env.OPENAI_API_KEY}`,
                'Content-Type': 'application/json'
            }
        }
    );

    return response.data.choices[0].message.content;
}

module.exports = { askJarvis };

File: jarvis_brain.py
import os
from openai import OpenAI
from dotenv import load_dotenv
from conversation_memory import load_memory


load_dotenv()
client = OpenAI(api_key=os.getenv("OPENAI_API_KEY"))

def ask_jarvis(prompt):
    memory = load_memory()
    recent = memory[-5:]  # Get last 5 interactions

    context = [
        {"role": "system", "content": "You are JARVIS, a helpful AI assistant loyal to Ra."}
    ]

    # Add recent memory to the prompt
    for convo in recent:
        context.append({"role": "user", "content": convo["user"]})
        context.append({"role": "assistant", "content": convo["jarvis"]})

    # Add the new message
    context.append({"role": "user", "content": prompt})

    response = client.chat.completions.create(
        model="gpt-3.5-turbo",
        messages=context
    )

    return response.choices[0].message.content
